### PR TITLE
[comfoair] Re-implement device options as channels

### PR DESCRIPTION
**Pull request by boehan**
Re-implemented ComfoAir device options as channels (from thing properties) to make them configurable [as requested from users](https://community.openhab.org/t/comfoair-new-comfoair-binding/92891/44?u=boehan) and [suggested by Hilbrand](https://community.openhab.org/t/dynamic-thing-configuration-options/106461/2?u=boehan).
Thus, also dynamic channel configuration based on the previous thing properties becomes obsolete.

I also renamed `ewt` (for "Erdwärmetauscher" in german) channels to `ghx` (geothermal heat exchanger) and `analog_priority` channel was moved to `options` channel group. Even though these changes were not required, I thought it would be best chance to do it now before OH3 since this is a breaking change. Not sure if this has to be stated anywhere else to notify users?

### CODE REVIEW

**Comment by fwolter** [comment]
Are you sure these breaking changes are really necessary? Although breaking changes could be made when moving to OH3, they are still anoying to any user.

**Comment by boehan** [comment]
> Are you sure these breaking changes are really necessary? Although breaking changes could be made when moving to OH3, they are still anoying to any user.

Renaming of the `ewt` channels ist not necessary from a technical perspective, I just thought it would be more self descriptive for non-german users (even though they were also named `ewt` in the v1 binding). Since the v2 binding is rather new (in the repo since 2.5.9) I think the affected number of users is still pretty small, especially as it affects only those who have the appropriate option installed. (The impact would be even smaller if the changes are cherry-picked to 2.5.x, though I'm aware that this is more than a bugfix). But I'm also okay with undoing this change again.
The other breaking change - moving `analogPriority` from `analogRF` to `options` group - should be done from a usability point of view, since it's in the same "menu" in the native control as well and has pretty much nothing to do with the RF controls. However, I'm pretty sure that this channel is hardly used by now.

**Comment by fwolter** [comment]
I agree that the changes make sense. But I as a user would be annoyed if I'm forced to change my config due to an unnecessary change under the hood. Bear in mind that OH3 is released only in two months.

**Comment by boehan** [comment]
As said, I'm also okay with leaving the `ewt` channels as is. Since this basically about naming conventions I would leave the decision to you and the other maintainers.

**Comment by fwolter** [comment]
My recommendation is to leave it as is.

**Comment by boehan** [comment]
I went with omitting channel renaming and thus breaking changes for the users (according to `ewt` channels), but harmonized and "internationalized" internal variables for consistency.

**Comment by boehan** [comment]
somehow can't get the signoff of the last commit fixed even though it's in the commit message...

**Comment by fwolter** [comment]
The revert commit doesn't contain a sign-off. You can click on the DCO check to view the failing commit and retrieve the necessary commands to fix it.

**Comment by boehan** [comment]
> The revert commit doesn't contain a sign-off. You can click on the DCO check to view the failing commit and retrieve the necessary commands to fix it.

Tried with the DCO check suggestion a few times, but that didn't help. But got it fixed now anyway.

File: bundles/org.openhab.binding.comfoair/README.md
Comment (fwolter) [inline bundles/org.openhab.binding.comfoair/README.md:165]: Can these change over time? If they are fixed, they should be properties rather than Channels.

File: bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fwolter) [inline bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml:857]: The readme states, these are read only.

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.comfoair/README.md
Comment (boehan) [reply bundles/org.openhab.binding.comfoair/README.md:165]: Unlike my assumption, and thus the initial implementation as properties, it seems to be quite common that individual device options are swapped or deactivated during operation (as communicated by [users](https://community.openhab.org/t/comfoair-new-comfoair-binding/92891/44?u=boehan)).

File: bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml
Comment (boehan) [reply bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml:857]: I'm not sure if I understand your comment correctly, but as stated above these channels are NOT meant to be read only

File: bundles/org.openhab.binding.comfoair/README.md
Comment (fwolter) [reply bundles/org.openhab.binding.comfoair/README.md:165]: Ok, fair enough.

File: bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fwolter) [reply bundles/org.openhab.binding.comfoair/src/main/resources/OH-INF/thing/thing-types.xml:857]: Please ignore my comment, was a bit to late yesterday.

**Review by fwolter — APPROVED** [review]
LGTM

**Pull request merged**
